File: src/Data Structure/Tree/Binary Tree/In Order Traversal Binary Tree.cpp
#include <bits/stdc++.h>
using namespace std;

struct tree
{
  char data;
  tree *left;
  tree *right;
};

typedef struct tree node;

void InOrder(node *root)
{
  if (root->left != NULL)
    InOrder(root->left);
  printf("%c", root->data);
  if (root->right != NULL)
    InOrder(root->right);
}
void CreateTree(node *root)
{
  char ans;

  fflush(stdout);
  printf("%c has any left child? ", root->data);

  fflush(stdin);
  scanf("%c", &ans);

  if (toupper(ans) == 'Y')
  {
    root->left = new node();
    fflush(stdout);
    printf("Enter Left Child Data of %c: ", root->data);
    fflush(stdin);
    scanf("%c", &root->left->data);
  }
  else
    root->left = NULL;
  fflush(stdout);
  printf("%c has any right child?: ", root->data);
  fflush(stdin);
  scanf("%c", &ans);
  if (toupper(ans) == 'Y')
  {
    root->right = new node();
    fflush(stdout);
    printf("Enter Right Child Data of %c: ", root->data);
    fflush(stdin);
    scanf("%c", &root->right->data);
  }
  else
    root->right = NULL;

  CreateTree(root->left);
  CreateTree(root->right);
}
int main()
{
  node *root = new node();
  printf("Enter Root Data: ");
  scanf("%c", &root->data);
  CreateTree(root);
  InOrder(root);
  return 0;
}

File: src/Data_Structure/Tree/Binary_Tree/In_Order_Traversal_Binary_Tree.java
package Data_Structure.Tree.Binary_Tree;

import java.util.Scanner;

public class In_Order_Traversal_Binary_Tree {
  static void inOrder(Node root) {
    if (root.left != null) {
      inOrder(root.left);
    }
    System.out.printf("%s", root.data);
    if (root.right != null) {
      inOrder(root.right);
    }
  }

  static void createTree(Node root) {
    if (root == null) return;

    Scanner sc = new Scanner(System.in);

    String ans;
    System.out.printf("%s has any left child? ", root.data);

    ans = sc.next();

    if (ans.equalsIgnoreCase("Y")) {
      root.left = new Node();
      System.out.printf("Enter left child data of %s: ", root.data);
      root.left.data = sc.next();
    } else {
      root.left = null;
    }

    System.out.printf("%s has any right child?: ", root.data);
    ans = sc.next();
    if (ans.equalsIgnoreCase("Y")) {
      root.right = new Node();
      System.out.printf("Enter right child data of %s: ", root.data);
      root.right.data = sc.next();
    } else {
      root.right = null;
    }

    createTree(root.left);
    createTree(root.right);
  }

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);

    Node root = new Node();
    System.out.printf("Enter root data: ");
    root.data = sc.next();

    createTree(root);
    inOrder(root);

    return;
  }
}
